iceoryx_posh/include/iceoryx_posh/internal/roudi/process.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_POSH_ROUDI_PROCESS_HPP
+#define IOX_POSH_ROUDI_PROCESS_HPP
+
+#include "iceoryx_posh/internal/mepoo/segment_manager.hpp"
+#include "iceoryx_posh/internal/roudi/introspection/process_introspection.hpp"
+#include "iceoryx_posh/internal/roudi/port_manager.hpp"
+#include "iceoryx_posh/internal/runtime/ipc_interface_user.hpp"
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+#include "iceoryx_posh/version/compatibility_check_level.hpp"
+#include "iceoryx_posh/version/version_info.hpp"
+#include "iceoryx_utils/cxx/list.hpp"
+#include "iceoryx_utils/error_handling/error_handling.hpp"
+#include "iceoryx_utils/posix_wrapper/posix_access_rights.hpp"
+
+#include <cstdint>
+#include <ctime>
+
+namespace iox
+{
+namespace roudi
+{
+class Process
+{
+  public:
+    /// @brief This class represents an application which has registered at RouDi and manages the communication to the
+    /// application
+    /// @param [in] name of the process; this is equal to the IPC channel name, which is used for communication
+    /// @param [in] pid is the host system process id
+    /// @param [in] payloadMemoryManager is a pointer to the payload memory manager for this process
+    /// @param [in] isMonitored indicates if the process should be monitored for being alive
+    /// @param [in] payloadSegmentId is an identifier for the shm payload segment
+    /// @param [in] sessionId is an ID generated by RouDi to prevent sending outdated IPC channel transmission
+    Process(const ProcessName_t& name,
+            int32_t pid,
```

Comment:
Done. AFAICT `const mepoo::MemoryManager` (const data) is not possible because of `ChunkSenderData(cxx::not_null<mepoo::MemoryManager* const> memoryManager,..)`